AdyenCard/Components/GiftCardComponent/GiftCardComponent.swift

```diff
@@ -263,38 +302,59 @@ public final class GiftCardComponent: PartialPaymentComponent,
         delegate.showConfirmation(for: component)
     }
 
-    private func requestOrder() -> Result<Void, Swift.Error> {
+    // MARK: - Partial Payment flow
+
+    private func startPartialPaymentFlow(paymentData: PaymentComponentData) -> Result<Void, Swift.Error> {
         if let order = order {
-            submit(order: order)
+            submit(order: order, paymentData: paymentData)
             return .success(())
         }
         guard let partialPaymentDelegate = partialPaymentDelegate else {
             AdyenAssertion.assertionFailure(message: Error.missingPartialPaymentDelegate.localizedDescription)
             return .failure(Error.missingPartialPaymentDelegate)
         }
         partialPaymentDelegate.requestOrder { [weak self] result in
-            self?.handle(orderResult: result)
+            self?.handle(orderResult: result, paymentData: paymentData)
         }
         return .success(())
     }
 
-    private func handle(orderResult: Result<PartialPaymentOrder, Swift.Error>) {
-        orderResult.handle(success: submit(order:), failure: { delegate?.didFail(with: $0, from: self) })
+    private func handle(orderResult: Result<PartialPaymentOrder, Swift.Error>,
+                        paymentData: PaymentComponentData) {
+        orderResult.handle(success: {
+            submit(order: $0, paymentData: paymentData)
+        }, failure: {
+            delegate?.didFail(with: $0, from: self)
```

Comment:
can we call `handle(error: $0)` here? so that all errors are handled in one place.

Reply:
👍 good one